Moved data to where it is most relevant

ConnorBarroso opened the pull request:
(I forgot to remove comments I left for myself)

App.js:
Removed fetchData from App
Currency is used everywhere so I keep it in the App
removed Symbol and string from currency
Removed All fetchData related assignments in compnentDidMount
Added localCurrency assignment to componentDidMount
Added handleCurrency callback function
pass handleCurrency into NavBar
only pass currency to CoinsPage

CurrencyDropdown.js:
made it a class component
setting the currency symbol to state
Reworked handleOnChange so symbols are matched to the target value, then pass the target value to the handleCurrency function.

ListedCoin.js:
replace currencyString with currency

NavBar.js:
Only passing currency and handleCurrency to CurrencyDropdown

ProgressBar.js: 
Fixed a bug where 0 would not be returned. 

TableDisplay:
Restructured imports to be node then relative
Removed all references to my own infinite scroll function
Renamed my state items to be more easy to understand
Moved all fetchData to TableDisplay
added sortLocal state to determine if the displayed list is showing the locally sorted version or the Api provided version. 
Made addColors an external function that adds colors into the listed coin properties.
rebuilt my componentDidMount and componentDidUpdate
due to the page state being within the fetchData state object I cant use just fetchData in my componentDidUpdate
moved my handleFetchData function from App.js to here. This function set sortLocal to false so if you click the sort by volume button it will change to be the API provided list. 
Added a handleNext function that adds the new items to the end of the list. 
I set the displaytedList to bne either the direct list state or the sorted version based off the sortLocal state.
Added infiniteScroll component

TableDisplay.styles.js:
Reworked it to your specs

CoinsPage.js/ styles:
Moved relevant data lower down the tree. 

utils.js: 
changed some variable names to reflect new versions.

ConnorBarroso committed:
Moved data to where it is most relevant

yoloOnTheBattlefield commented on src/App.js:
```diff
@@ -9,65 +9,33 @@ import {
   render() {
-    const { toggle, fetchData } = this.state;
-    const { currency } = fetchData;
+    const currency = this.state.currency;
```
you can destructure here

yoloOnTheBattlefield replied:
you can destructure here

ConnorBarroso committed:
graphs fix, Currency changes, Infinite scroll test

ConnorBarroso commented:
Push 2:
**GraphDisplay.js**: As this component is the parent of both graphs that use the same API request, I moved that request here to remove a redundant request. 
I added in the day select variable and buttons, the day Data is held both in state and local storage to preserve day selected between refreshes. 
I added in both Mobile and Desktop displays that render depending on max-width breakpoints. Though the Carousel I use is **not fully implemented.** 

**LineGraph.js**: Rebuilt the LineGraph component receiving the relevant API data and currency as props.
Formatted price and dates accordingly. 
added in the new method for the background gradient as we went through in the meeting today. 
Added labels to the graph

**BarGraph.js**: Implemented the Bar Graph in the same fashion as the Line Graph
Implemented the  price and date formatters for proper display. 
Adjusted the border width to keep bars visible with larger data arrays. 

**CurrencyDropdown.js**: After discovering the built in currency format function, I rebuilt my currency dropdown to be more lightweight and straightforward. 
The currency and handleCurrency function is passed in from App.js. 
The handleOnChange function sets the correct currency symbol then changes the currency using handleCurrency.

**HeaderInfo.js**: added the progressBar component to display the market cap percentage  for bitcoin and Ethereum correctly. 

**ProgressBar.js**:Added a compact prop that determines if the labels are displayed. 

**TableDisplay.js**: implemented the infinite loader but only in the **very early stages**. 

**TableGraph.js**: Added and then removed changes to make the graphs as lightweight as possible. (caused a bug that crashed the page when the page was widened)


**utils.js**: added optional chaining to avoid undefined errors

yoloOnTheBattlefield commented on src/components/CurrencyDropdown/CurrencyDropdown.js:
```diff
@@ -1,32 +1,36 @@
       cad: "$",
       eur: "€",
       gbp: "£",
     };
```
you can put this outside the class

yoloOnTheBattlefield commented on src/components/CurrencyDropdown/CurrencyDropdown.js:
```diff
@@ -1,32 +1,36 @@
-};
+  render() {
+    const { currency } = this.props;
+    const currencyArray = ["usd", "cad", "eur", "gbp"];
```
this too, or you could convert the object into an array and map over it. DRY

yoloOnTheBattlefield commented on src/components/CurrencyDropdown/CurrencyDropdown.js:
```diff
@@ -1,32 +1,36 @@
+        <Symbol>{this.state.symbol}</Symbol>
+        <StyledSelect onChange={this.handleOnChange} value={currency}>
+          {currencyArray.map((i) => (
+            <option value={i} key={i}>
```
don't use index as key

yoloOnTheBattlefield commented on src/components/TableDisplay/TableDisplay.js:
```diff
@@ -12,135 +23,252 @@ import {
+              />
+            ))}
+          </InfiniteScroll>
+        </InfiniteScrollContainer>} */}
```
do you need this?

ConnorBarroso replied:
I'm not sure, handling the infinite scroll will be my next big task

ConnorBarroso committed:
implementing suggested fixes

ConnorBarroso commented:
Implemented the suggested changes
